temporalio/client.py

```diff
@@ -3664,6 +3921,96 @@ async def __anext__(self) -> ScheduleListDescription:
             return ret
 
 
+class WorkflowUpdateHandle(Generic[LocalReturnType]):
+    """Handle for a workflow update execution request."""
+
+    def __init__(
+        self,
+        client: Client,
+        id: str,
+        name: str,
+        workflow_id: str,
+        *,
+        workflow_run_id: Optional[str] = None,
+        result_type: Optional[Type] = None,
+    ):
+        """Create a workflow update handle.
+
+        Users should not create this directly, but rather use
+        :py:meth:`Client.start_workflow_update`.
+        """
+        self._client = client
+        self._id = id
+        self._name = name
+        self._workflow_id = workflow_id
+        self._workflow_run_id = workflow_run_id
+        self._result_type = result_type
+        self._known_result: Optional[temporalio.api.update.v1.Outcome] = None
+
+    @property
+    def id(self) -> str:
+        """ID of this Update request."""
+        return self._id
+
+    @property
+    def name(self) -> str:
+        """The name of the Update being invoked."""
+        return self._name
+
+    @property
+    def workflow_id(self) -> str:
+        """The ID of the Workflow targeted by this Update."""
+        return self._workflow_id
+
+    @property
+    def workflow_run_id(self) -> Optional[str]:
+        """If specified, the specific run of the Workflow targeted by this Update."""
+        return self._workflow_run_id
+
+    async def result(
+        self,
+        *,
+        timeout: Optional[timedelta] = None,
+        rpc_metadata: Mapping[str, str] = {},
+        rpc_timeout: Optional[timedelta] = None,
+    ) -> LocalReturnType:
+        """Wait for and return the result of the update. The result may already be known in which case no call is made.
+        Otherwise the result will be polled for until returned, or until the provided timeout is reached, if specified.
+
+        Args:
+            timeout: Optional timeout specifying maximum wait time for the result.
+            rpc_metadata: Headers used on the RPC call. Keys here override client-level RPC metadata keys.
+            rpc_timeout: Optional RPC deadline to set for the RPC call. If this elapses, the poll is retried until the
+                overall timeout has been reached.
+
+        Raises:
+            TimeoutError: The specified timeout was reached when waiting for the update result.
+            RPCError: Update result could not be fetched for some other reason.
+        """
+        if self._known_result is not None:
+            outcome = self._known_result
+            return await _update_outcome_to_result(
+                outcome,
```

Comment:
.NET did call it `KnownOutcome` but I don't think it matters much since this is private API. It seems ok to me to name something "known result" if it's a "known" thing for the "result" method later.

Reply:
Renamed